src/testSession.ts

```diff
@@ -111,14 +115,15 @@ export class TestSession {
     // Write the test session options used to create this session
     fsCore.writeJsonSync(path.join(this.dir, 'testSessionOptions.json'), JSON.parse(JSON.stringify(options)));
 
+    const authStrategy = options.authStrategy ? AuthStrategy[options.authStrategy] : undefined;
     // have to grab this before we change the home
-    transferExistingAuthToEnv();
+    transferExistingAuthToEnv(authStrategy);
 
     // Set the homedir used by this test, on the TestSession and the process
     // TODO: does this work on Windows?
     process.env.HOME = this.homeDir = env.getString('TESTKIT_HOMEDIR', this.dir);
     process.env.SFDX_USE_GENERIC_UNIX_KEYCHAIN = 'true';
-    testkitHubAuth(this.homeDir);
+    testkitHubAuth(this.homeDir, authStrategy);
```

Comment:
@mshanemc - rather than do this `pre-auth` then homedir stub then `post-auth` flow, what if the testkit authenticator handled the homedir stubbing as well?  That would make this all just 1 call.  Not a big deal since this is all in TestSession internals but just curious if it would make things easier or cleaner.